crates/graphql_artifact_generation/src/generate_artifacts.rs

```diff
@@ -90,123 +91,93 @@ pub fn get_artifact_path_and_content(
     }
 
     for (
-        encountered_client_field_id,
-        ClientFieldTraversalResult {
+        encountered_field_id,
+        FieldTraversalResult {
             traversal_state,
             merged_selection_map,
             was_ever_selected_loadably,
             ..
         },
     ) in &global_client_field_map
     {
-        let encountered_client_field = schema.client_field(*encountered_client_field_id);
-        // Generate reader ASTs for all encountered client fields, which may be reader or refetch reader
-        match &encountered_client_field.variant {
-            ClientFieldVariant::UserWritten(info) => {
-                path_and_contents.extend(generate_eager_reader_artifacts(
+        match encountered_field_id {
+            FieldType::ServerField(encountered_server_field_id) => {
+                let encountered_server_field = &schema.server_field(*encountered_server_field_id);
+
+                // path_and_contents.push(generate_refetch_reader_artifact(
+                //     schema,
+                //     encountered_server_field,
+                //     None,
+                //     &traversal_state.refetch_paths,
+                //     true,
+                // ));
+
+                generate_entrypoint_artifacts_with_field_traversal_result(
                     schema,
-                    encountered_client_field,
-                    project_root,
-                    artifact_directory,
-                    *info,
-                    &traversal_state.refetch_paths,
-                ));
-
-                if *was_ever_selected_loadably {
-                    path_and_contents.push(generate_refetch_reader_artifact(
-                        schema,
```

Comment:
3. All of this gets extracted into `generate_entrypoint_artifacts_with_field_traversal_result`